packages/monaco/src/browser/monaco-editor-provider.ts

```diff
@@ -35,11 +35,22 @@ import { MonacoBulkEditService } from './monaco-bulk-edit-service';
 
 import IEditorOverrideServices = monaco.editor.IEditorOverrideServices;
 import { ApplicationServer } from '@theia/core/lib/common/application-protocol';
-import { OS } from '@theia/core';
+import { OS, MaybePromise, ContributionProvider, Prioritizeable } from '@theia/core';
 import { KeybindingRegistry, OpenerService, open, WidgetOpenerOptions } from '@theia/core/lib/browser';
 import { MonacoResolvedKeybinding } from './monaco-resolved-keybinding';
 import { HttpOpenHandlerOptions } from '@theia/core/lib/browser/http-open-handler';
 
+export const MonacoEditorOptionsProvider = Symbol('MonacoEditorOptionsProvider');
```

Comment:
I would prefer to have one contribution which is scheme based instead of priotirizable. I don't think there will be many use cases and they can be resolved by `scheme` only. Otherwise we introduce new contribution point in ad-hoc manner to satisfy a single use case. It would be better to move in direction of some simple design, something like:
```ts
export interface MonacoEditorFactory {
   readonly scheme: string;
   create(model: MonacoEditorModel, defaultsOptions: MonacoEditor.IOptions, defaultServices: IEditorOverrideServices): MonacoEditor;
}

@injectable()
export class MonacoEditorProvider {
    @inject(ContributionProvider)
    @named(MonacoEditorFactory)
    protected readonly factories: ContributionProvider<MonacoEditorFactory>;
}
```

`MonacoEditorFactory` could be an abstract class not an interface providing template methods to override scheme, options and services for subclasses.

If there is a middle ground for now which does not require much refactorings but moves in direction of design above i would be fine too.

Reply:
I made these changes, thanks for the pointer. :+1: